crates/ironrdp-async/src/framed.rs

```diff
@@ -164,23 +168,22 @@ where
     /// `tokio::select!` statement and some other branch
     /// completes first, then it is safe to drop the future and re-create it later.
     /// Data may have been read, but it will be stored in the internal buffer.
-    pub async fn read_by_hint(&mut self, hint: &dyn PduHint) -> io::Result<Bytes> {
+    pub async fn read_by_hint(&mut self, hint: &dyn PduHint) -> io::Result<(Bytes, Vec<Bytes>)> {
```

Comment:
suggestion: I feel it would be better to follow the same pattern everywhere, and optionally provide the accumulator like this.
```suggestion
    pub async fn read_by_hint(&mut self, hint: &dyn PduHint, unmatched: Option<&mut Vec<Bytes>>) -> io::Result<Bytes> {
```

We can just drop the unmatched frame when the accumulator is `None`.

Reply:
ok